### PR TITLE
Add AccountId and CredentialScope BuiltIns

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
Add AccountId and CredentialScope BuiltIns

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source-2.0/aws/rules-engine/built-ins.rst [hunk 1]
```diff
@@ -68,6 +68,25 @@ Description
 Type
     ``boolean``
 
+.. _rules-engine-aws-built-ins-account-id:
+
+``AWS::Auth::AccountId`` built-in
+=========================
```
Comment (kstich) [inline docs/source-2.0/aws/rules-engine/built-ins.rst:74]: These header formatting pieces on both of these need to match the length of the header they're for.

File: smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    .getEndpointRuleSet()));
+        }
+        return events;
+    }
+
+    private List<ValidationEvent> validateRuleSetAwsBuiltIns(ServiceShape serviceShape, EndpointRuleSet ruleSet) {
+        List<ValidationEvent> events = new ArrayList<>();
+        for (Parameter parameter : ruleSet.getParameters()) {
+            if (parameter.isBuiltIn()) {
+                validateBuiltIn(serviceShape, parameter.getBuiltIn().get(), parameter, "RuleSet")
```
Comment (kstich) [inline smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java:48]: Could this use "BuiltIn" and then include the name in the final list for the event?

File: smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    private Optional<ValidationEvent> validateBuiltIn(
+            ServiceShape serviceShape,
+            String builtInName,
+            FromSourceLocation source,
+            String... eventIdSuffixes
+    ) {
+        if (ADDITIONAL_APPROVAL_BUILT_INS.contains(builtInName)) {
+            return Optional.of(danger(serviceShape, source, String.format(
```
Comment (kstich) [inline smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java:62]: Can you add an `errorfiles` test case that shows these being triggered?

**Review by kstich — CHANGES_REQUESTED** [review]

File: docs/source-2.0/aws/rules-engine/built-ins.rst (hunk 1 above)
Comment (alextwoods) [reply docs/source-2.0/aws/rules-engine/built-ins.rst:74]: Thanks - forgot to run `make html` - fixed (and fixed the duplicate account-id label).

File: smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java (hunk 2 above)
Comment (alextwoods) [reply smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java:48]: Thought I could do this - but allows easy by passing of the validator by changing one of the fields used in equality checks (eg, documentation).

File: smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java (hunk 3 above)
Comment (alextwoods) [reply smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java:62]: Good call - added.

File: smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        return events;
+    }
+
+    private Optional<ValidationEvent> validateBuiltIn(
+            ServiceShape serviceShape,
+            String builtInName,
+            FromSourceLocation source,
+            String... eventIdSuffixes
+    ) {
+        if (ADDITIONAL_APPROVAL_BUILT_INS.contains(builtInName) || builtInName.equals("AccountId")) {
```
Comment (kstich) [inline smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java:61]:
```suggestion
        if (ADDITIONAL_APPROVAL_BUILT_INS.contains(builtInName)) {
            List<String> suffixes = new ArrayList(ListUtils.of(eventIdSuffixes));
            suffixes.add(builtInName);
```

File: smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            return Optional.of(danger(serviceShape, source, String.format(
+                            "The `%s` built-in used requires additional consideration of the rules that use it.",
+                            builtInName),
+                    String.join(".", Arrays.asList(eventIdSuffixes))));
```
Comment (kstich) [inline smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/RuleSetAwsBuiltInValidator.java:65]:
```suggestion
                    String.join(".", suffixes)));
```

**Review by kstich — APPROVED** [review]

**Pull request merged**